### PR TITLE
Baton overhaul.

### PR DESCRIPTION
**Pull request by Tunguso4ka**
## About the PR

Adds telescopic baton to the sectech and to the command roles backpacks. Adds police baton for the detective in the detdrobe and detective closet. Brings /tg/station variant of stunbaton textures.

**Media**

зображення

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Tunguso4ka
- add: adds telescopic and police batons to the station.

### CODE REVIEW

**Comment by Tunguso4ka** [comment]
i redo some of this.

**Comment by Tunguso4ka** [comment]
It is done!

**Comment by Flareguy** [comment]
Not a big fan of the /tg/ stunbaton sprites, they aren't really as cool in my opinion.

And on the balance side: the disarm malus is unnessecary and cringe, and the lack of any extended attack time basically makes these things almost as good as a stunbaton. In my opinion, they should be able to subdue an unruly tider with a bit of effort, but suck as an actual weapon. Otherwise you get something like the lame ass one from SS13 that basically just instantly ends any combat encounter if they land two consecutive hits.

File: Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    [DataField("onThrowStunChance")]
+    public float onThrowStunChance { get; set; } = 0.20f;
+
+    [DataField("stunSound")]
+    public SoundSpecifier StunSound { get; set; } = new SoundPathSpecifier("/Audio/Weapons/egloves.ogg");
+
+    [DataField("extendSound")]
+    public SoundSpecifier ExtendSound { get; set; } = new SoundPathSpecifier("/Audio/Weapons/batonextend.ogg");
+  }
+}
```
Comment (metalgearsloth) [inline Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs:23]: Components shouldn't be singleton.
1. Stun baton, telebaton, and esword should be using a generic toggleable component.
2. Stun baton and telebaton should be using the same code.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs (hunk 1 above)
Comment (Tunguso4ka) [reply Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs:23]: will do.

Comment (Tunguso4ka) [reply Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs]: i tried combining it all in one component, but it seems that better solution is just let it be like it been i believe.

File: Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs
Comment (AJCM-git) [inline Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs:1]: Formatting is weird, there is only 1 space between the fields and the braces indentation, also you should be using file scoped namespaces (google for examples)
A bigger issue is that this should just share code with the stun baton

File: Content.Server/Weapons/Melee/Telebaton/TelebatonSystem.cs
Comment (AJCM-git) [inline Content.Server/Weapons/Melee/Telebaton/TelebatonSystem.cs:1]: Same formatting issues
This should also just share code with stun batons and its using obsolete methods like SoundSystem.Play and not using proxy methods when possible (there is a list of them in EntitySystem.Proxy)

File: Resources/Audio/Weapons/licenses.txt [hunk 2]
```diff
@@ -12,4 +12,6 @@ boxingbell.ogg taken from Herkules92 at https://freesound.org/people/Herkules92/
 
 block_metal1.ogg taken from https://github.com/Citadel-Station-13/Citadel-Station-13/commit/31c5996a5db8cce0cb431cb1dc20d99cac83f268 under CC BY-SA 3.0
 
-pierce.ogg taken from: https://github.com/tgstation/tgstation/commit/106cd26fc00851a51dd362f3131120318d848a53
\ No newline at end of file
+pierce.ogg taken from: https://github.com/tgstation/tgstation/commit/106cd26fc00851a51dd362f3131120318d848a53
+
+batonextend.ogg taken from https://github.com/tgstation/tgstation/commit/d4f678a1772007ff8d7eddd21cf7218c8e07bfc0
```
Comment (AJCM-git) [inline Resources/Audio/Weapons/licenses.txt:17]: make a rga for these (they are normally called attributes.yml or something, there are many examples of them in the repo)

File: Resources/Locale/en-US/weapons/melee/melee.ftl [hunk 3]
```diff
@@ -1 +1,3 @@
 melee-inject-failed-hardsuit = Your {$weapon} cannot inject through hardsuits!
+melee-telebaton-examined-extended = A compact and elegant tool for subjugating your staff when they feel like not taking orders. Activate it in your hand to retract.
+melee-telebaton-examined-retracted = A compact and elegant tool for subjugating your staff when they feel like not taking orders. Activate it in your hand to extend.
```
Comment (AJCM-git) [inline Resources/Locale/en-US/weapons/melee/melee.ftl:3]: These should just be set up like this ( the normal desc is already in the prototype)

```suggestion
melee-telebaton-examined-extended = Activate it in your hand to retract.

melee-telebaton-examined-retracted = Activate it in your hand to extend.
```

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by metalgearsloth** [comment]
@Tunguso4ka are you coming back to this?

**Comment by Tunguso4ka** [comment]
> @Tunguso4ka are you coming back to this?

yes, i will rework it in the next days.

**Pull request closed without merging**